### DISCUSSION TITLE
Random questions (step concept for lifepath, screen readers)

### DISCUSSION THREAD

**Discussion by Felderburg** [bcholmes/StarTrek2d20#121] [Q&A] [answered]
1. I saw a commit yesterday that stated it was to "begin applying a step concept to the lifepath character creation". What does that mean?
2. I just noticed the little random LCARS numbers at the footer of the page. I wanted to see how it was done, and discovered they're hidden from screen readers using the aria-hidden tag. Obviously these numbers aren't functional, but they are part of the flavor. Is there a way to get that flavor in somehow by indicating (voicing/printing in screen readers) that it's a random non-functional LCARS string of numbers? I don't know how important "flavor" is to people who use screen readers, but it might be worth finding out.

**Comment by bcholmes** [comment, marked as answer]
1. Conceptually, the lifepath "steps" say things like, "you can this many attribute points for this step in the process (with these choices), and this many discipline points, etc." In the current implementation, the internal guts of the character don't really "remember" the steps; it just remembers that "the total attributes are now X, and the total disciplines are now Y". That's okay, but it limits my ability to do certain things (maybe I just want to go back and change one step in the process. I can do that with "History", but the way it does that is pretty brute-force-y -- essentially keeping back-up copies of the character every time you move from one screen to another). I think that some of the things I want to do with the code will be more straight-forward if I just make use of steps.

2. I hadn't really thought about the idea of suppressing flavour. When I implemented that I had recently seen one of the Continuing Conversations videos about PDF layout and how some folks who use screen readers would get random numbers read out at them because they're decorative text on the PDFs. So I was trying to avoid that. If I hear from some screen-reader users, I'll revisit that idea.

**Reply by 1001100x02** [reply]
Yes, the ability to selectively retro edit a particular step instead of having to redo everything fro that point forwards would be a godsend!
